frontend/src/views/GSecrets.vue

```diff
@@ -525,8 +515,13 @@ export default {
     relatedShootCountDns (secret) {
       const shootsByDnsSecret = filter(this.shootList, shoot => {
         return some(shoot.spec.dns?.providers, {
-          type: secret.metadata.dnsProviderName,
           secretName: secret.metadata.name,
+        }) ||
```

Comment:
```js
    relatedShootCountDns (secret) {
      const hasDnsProvider = (providers, secretName) => some(providers, [
        ['secretName'], 
        secretName],
      )
      const hasSecretResource = (resources, secretName) => some(resources, {
        resourceRef: { 
           kind: 'Secret', 
           name: secretName,
        },
      })

      let count = 0
      for (const shoot of this.shootList) {
        const secretName = secret.metadata.name
        if (hasDnsProvider(shoot.spec.dns?.providers, secretName) || hasSecretResource(shoot.spec.resources, secretName)) {
          count++
        }
      }
      return count
    }
```
This function is called twice in the getter`dnsSecretItems`. It should be called only once. Same for `relatedShootCountInfra` which is called twice in `infrastructureSecretItems`.